sqlite/sqlite.go

```diff
@@ -230,14 +231,38 @@ func (s *SqlStore) RestoreSqlStore(ctx context.Context, r io.Reader) error {
 		return err
 	}
 
-	// Close the current DB.
+	// Close the current DB and run the restore while under lock.
+	s.Mu.Lock()
+	if err := s.runSqlRestore(ctx, tempFileName); err != nil {
+		s.Mu.Unlock()
+		return err
+	}
+
+	// Now that the DB has been re-opened, release the lock so that the migrations
+	// can be run.
+	s.Mu.Unlock()
```

Comment:
Same question as in the bolt code: are we open to a race condition here? And if so, could we refactor the migrator to avoid it?

Reply:
Addressed in the same way as with the bolt code...run the migrations on the restored database before switching over the active database to the restored database. Refactoring the sqlite migrator to not use a lock of its own would be easier than refactoring the bolt one, but this was even easier than that still.